packages/storybook/src/utilities/typed-object-keys.ts

```diff
@@ -0,0 +1,5 @@
+// Based on https://fettblog.eu/typescript-better-object-keys/
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface ObjectConstructor {
+    keys<T extends object>(o: T): (keyof T)[];
+}
```

Comment:
It's hard to rate these things but a solution that overrides how the keys property is defined globally is a little more of a hack than a single `as` cast. I think it's still worth trying to find an alternative to `as` casts but this is probably not the right direction to jump to first as a solution.

Reply:
I don't think there's much else we can do. I did a fair bit of digging, and unless there's some alternative to `Object.keys()`, I think that's it.
https://alexharri.com/blog/typescript-structural-typing
https://fettblog.eu/typescript-better-object-keys/
If you search the repo for `Object.keys`, you'll see we are casting the result in plenty of other places.

I've reverted back to the `as` cast per your preference.

Reply:
Sounds reasonable, thanks for looking into it